### PR TITLE
Add responsive table.

### LINKED ISSUE #525: Update responsive nature of Table Component.
## Problem

There is a need for an update in the responsive nature of the table component to make it more visible in smaller screens. As `table-container ` class from Bulma is not much of our use.
Referencing conversation - [Intern guide issue](https://github.com/creativecommons/creativecommons.github.io-source/pull/339#issuecomment-658979570) 
## Implementation

- [x] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by Dhruvi16**
## Fixes

Fixes #525 by @Dhruvi16 

## Description

This PR adds a responsive table.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Dhruvi16** [comment]
So far I have this -
image


There are a few problems with this -
1. The data is aligned right.
2. The stripes do not follow DL.

**Comment by Dhruvi16** [comment]
I was unable to fix the above problems, some tips will be nice. 
Thanks

File: src/stories/tables.stories.mdx [hunk 1]
```diff
@@ -17,32 +17,32 @@ import { figmaConfig } from './helpers'
           <th>State name</th>
           <th>Capital of the state</th>
           <th>Population</th>
-          <th>Total seats</th>
+          <th>Total seats</ththe>
```
Comment (zackkrida) [inline src/stories/tables.stories.mdx]: `</ththe>` looks like a typo

File: src/stories/tables.stories.mdx [hunk 2]
```diff
@@ -17,32 +17,32 @@ import { figmaConfig } from './helpers'
...
       </thead>
       <tr>
-        <td>Alabama</td>
-        <td>Montgomery</td>
-        <td class="number-cell">4,903,185</td>
-        <td class="number-cell">7</td>
+        <td data-content="State name">Alabama</td>
+        <td data-content="Capital of the state">Montgomery</td>
+        <td data-content="Population" class="number-cell">4,903,185</td>
+        <td data-content="Total seats" class="number-cell">7</td>
```
Comment (zackkrida) [inline src/stories/tables.stories.mdx]: 😻 i LOVE how you implemented this. Awesome!

File: src/styles/table.scss [hunk 3]
```diff
@@ -2,9 +2,43 @@ $table-cell-padding: $space-normal;
...
+    line-height: 1.5;
+    height: 100%;
+    width: 45%;
+    text-align: left;
+    padding: $space-normal;
+    background-color: $color-light-gray;
+    border-right: 0.0625rem solid $color-light-gray;
+	}
+
+	td:nth-of-type(n):before { content: attr(data-content); }
```
Comment (zackkrida) [inline src/styles/table.scss]: 🔥 Great

**Review by zackkrida — CHANGES_REQUESTED** [review]

File: src/stories/tables.stories.mdx (hunk 2 above)
Comment (Dhruvi16) [reply src/stories/tables.stories.mdx]: I think I should change it to `data-label` because that makes more sense. What do you think?

Comment (Dhruvi16) [reply src/stories/tables.stories.mdx]: And it was @dhruvkb's idea so thanks to him.

Comment (zackkrida) [reply src/stories/tables.stories.mdx]: `data-label` probably makes sense!

**Comment by Dhruvi16** [comment]
I also have one doubt that merging this will break the existing tables in different projects because `data-label` will be missing there.

**Comment by zackkrida** [comment]
@Dhruvi16 I found two more issues:

1. Can you update _all_ table stories in vocabulary to have the `data-label`s?The 2nd table example is missing them.
2. The table rows need some border-radius and bottom margin between them:

Screen Shot 2020-08-12 at 12 46 01 PM

**Comment by Dhruvi16** [comment]
Okay! I recognize my mistake here but we discussed about having two types of responsive designs right?

**Comment by zackkrida** [comment]
@Dhruvi16 you're right, probably need a new top-level css class like `.is-responsive` or something for the tables that will have the `data-label` styles. Without that class it will use the "old"/overflow responsive styles

**Comment by Dhruvi16** [comment]
you are right! I will update it.

**Comment by Dhruvi16** [comment]
I think I have fixed all the problems. I am just not sure about the code.
image

**Review by zackkrida — APPROVED** [review]
This is great, thanks @Dhruvi16

**Pull request merged**